lib/datasets/job.rb

```diff
@@ -29,5 +29,17 @@ def self.deserialize(*args)
       raise NotImplementedError
     end
 
+    private
+
+    # Reinvent HashWithIndifferentAccess
+    def self.deserialize_volume(hash)
+      Volume.new(
+        namespace: hash[:namespace] || hash["namespace"],
+        id: hash[:id] || hash["id"],
+        access_profile: hash[:access_profile] || hash["access_profile"],
+        right: hash[:right] || hash["right"]
```

Comment:
Agreed, although I missed the map-reduce implementation.  I considered bringing in HashWithIndifferentAccess from active support, but for such a small scope this seemed fine.